pyspark/bigdl/optim/optimizer.py

```diff
@@ -732,8 +733,12 @@ def set_validation(self, batch_size, val_rdd, trigger, val_method=None):
         """
         if val_method is None:
             val_method = [Top1Accuracy()]
-        callBigDlFunc(self.bigdl_type, "setValidation", self.value, batch_size,
+        if isinstance(val_rdd, RDD):
```

Comment:
if else should only responsible for the method name instead of the entire callBigDLFunc